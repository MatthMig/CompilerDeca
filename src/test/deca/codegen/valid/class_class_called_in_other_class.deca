//  Description : 
// 
//      Test correct qui utilise une classe dans une autre classe
// 
//  Résultats :
// 
//      N/A 
// 
//  Historique : 
// 
//      Créé le 13/06 par gl03

class A {
    int x;
}

class B extends A {
    int methCmp0(){
        if(this.x<0)
        {
            print("Le nombre est inférieur à 0 : ");
        }
        else {
            print("le nombre est supérieur ou égal à 0 :");
        }
        return x;
    }
}

{
    B oB = new B();
    oB.x=0;
    println(oB.methCmp0());
}
