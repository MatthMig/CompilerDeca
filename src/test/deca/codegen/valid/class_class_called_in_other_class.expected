le nombre est supérieur ou égal à 0 :0